src/compile/data/index.ts

```diff
@@ -43,7 +44,28 @@ export interface DataComponent {
   isFaceted: boolean;
 
   /**
-   * Parse properties passed down from ancestors.
+   * Parse properties passed down from ancestors. Helps us to keep track of what has been parsed or is derived.
    */
-  ancestorParse: Dict<string>;
+  ancestorParse?: AncestorParse;
+}
+
+/**
+ * Class to track interesting properties (see https://15721.courses.cs.cmu.edu/spring2016/papers/graefe-ieee1995.pdf)
+ * about how fields have been parsed or whether they have been derived in a transforms. We use this to not parse the
+ * same field again (or differently).
+ */
+export class AncestorParse extends Split<Dict<string>> {
```

Comment:
If you want to rename it later, better rename it in this PR as this class is introduced in the PR.

Reply:
I like the name.